e2e/reboot_test.go

```diff
@@ -76,3 +80,35 @@ func nodesReady(c kubernetes.Interface, expectedNodes *v1.NodeList, t *testing.T
 		return nil
 	})
 }
+
+const checkpointAnnotation = "checkpointer.alpha.coreos.com/checkpoint-of"
+
+// controlPlaneReady waits for API server availability and no checkpointed pods
+// in kube-system.
+func controlPlaneReady(c kubernetes.Interface, attempts int, backoff time.Duration) error {
+	return retry(atttempts, backoff, func() error {
+		pods, err := c.CoreV1().Pods("kube-system").List(metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("get pods in kube-system: %v", err)
+		}
+
+		// list of pods that are checkpoint pods, not the real pods.
+		var (
+			checkpointPods []string
+			regularPods    []string
+		)
+		for _, pod := range pods.Items {
+			if len(pod.Annotations) > 0 && pod.Annotations[checkpointAnnotation] != "" {
+				checkpointPods = append(checkpointPods, pod.Name)
+			} else {
+				regularPods = append(regularPods, pod.Name)
+			}
+		}
+		if len(checkpointPods) > 0 {
+			sort.Strings(checkpointPods)
+			sort.Strings(regularPods)
+			return fmt.Errorf("kube-system still has checkpoint pods=%q, non-checkpoint pods=%q", checkpointPods, regularPods)
```

Comment:
this error message might super long, and not easy to read?

Reply:
ack, will clean it up